reverse_engineering/helpers/sequenceHelper.js

```diff
@@ -0,0 +1,290 @@
+const SHARING_LINK_REGEX = /\s+LINK$/;
+
+/**
+ * @enum {string}
+ */
+const OPTION_VALUE = {
+  true: 'Y',
+  false: 'N',
+};
+
+/**
+ * @enum {string}
+ */
+const SEQUENCE_OPTION = {
+  cache: 'cache',
+  cycle: 'cycle',
+  extend: 'extend',
+  global: 'global',
+  keep: 'keep',
+  noCache: 'nocache',
+  noCycle: 'nocycle',
+  noExtend: 'noextend',
+  noKeep: 'nokeep',
+  noOrder: 'noorder',
+  noScale: 'noscale',
+  noShard: 'noshard',
+  order: 'order',
+  scale: 'scale',
+  scaleExtend: 'scaleExtend',
+  session: 'session',
+  shard: 'shard',
+  sharing: 'sharing',
+  type: 'type',
+};
+
+const SEQUENCE_OPTION_MAP = {
+  [SEQUENCE_OPTION.cycle]: {
+    [OPTION_VALUE.true]: SEQUENCE_OPTION.cycle,
+    [OPTION_VALUE.false]: SEQUENCE_OPTION.noCycle,
+  },
+  [SEQUENCE_OPTION.order]: {
+    [OPTION_VALUE.true]: SEQUENCE_OPTION.order,
+    [OPTION_VALUE.false]: SEQUENCE_OPTION.noOrder,
+  },
+  [SEQUENCE_OPTION.shard]: {
+    [OPTION_VALUE.true]: SEQUENCE_OPTION.shard,
+    [OPTION_VALUE.false]: SEQUENCE_OPTION.noShard,
+  },
+  [SEQUENCE_OPTION.scale]: {
+    [OPTION_VALUE.true]: SEQUENCE_OPTION.scale,
+    [OPTION_VALUE.false]: SEQUENCE_OPTION.noScale,
+  },
+  [SEQUENCE_OPTION.type]: {
+    [OPTION_VALUE.true]: SEQUENCE_OPTION.session,
+    [OPTION_VALUE.false]: SEQUENCE_OPTION.global,
+  },
+  [SEQUENCE_OPTION.keep]: {
+    [OPTION_VALUE.true]: SEQUENCE_OPTION.keep,
+    [OPTION_VALUE.false]: SEQUENCE_OPTION.noKeep,
+  },
+  [SEQUENCE_OPTION.extend]: {
+    [OPTION_VALUE.true]: SEQUENCE_OPTION.extend,
+    [OPTION_VALUE.false]: SEQUENCE_OPTION.noExtend,
+  },
+};
+
+/**
+ * @param {{ schema: string, execute: Execute, logger: object }}
+ * @returns {Promise<SequenceDto[]>}
+ */
+const getSchemaSequenceDtos = async ({ schema, execute, logger }) => {
+  try {
+    const queryResult = await execute(
+      `
+      SELECT JSON_OBJECT(
+          'sharing'      VALUE SHARING,
+          'sequenceName' VALUE SEQUENCE_NAME,
+          'minValue'     VALUE MIN_VALUE,
+          'maxValue'     VALUE MAX_VALUE,
+          'increment'    VALUE INCREMENT_BY,
+          'cycle'        VALUE CYCLE_FLAG,
+          'order'        VALUE ORDER_FLAG,
+          'cacheValue'   VALUE CACHE_SIZE,
+          'scale'        VALUE SCALE_FLAG,
+          'extend'       VALUE EXTEND_FLAG,
+          'shard'        VALUE SHARDED_FLAG,
+          'type'         VALUE SESSION_FLAG,
+          'keep'         VALUE KEEP_VALUE
+      )
+      FROM ALL_OBJECTS
+      LEFT JOIN ALL_SEQUENCES
+                ON ALL_SEQUENCES.SEQUENCE_OWNER = ALL_OBJECTS.OWNER
+                AND ALL_SEQUENCES.SEQUENCE_NAME = ALL_OBJECTS.OBJECT_NAME
+      WHERE OBJECT_TYPE = 'SEQUENCE'
+      AND OWNER = '${schema}'
+      `
+    );
+
+    if (!queryResult?.length) {
+      return [];
+    }
+
+    const ddlScriptMap = await getSchemaSequenceDdl({
+      schema,
+      execute,
+      logger,
+    });
+
+    return queryResult.map(([data]) => {
```

Comment:
so here we are expecting to traverse through the `sequences`? Let's use `rawSequence` instead `data`. And on the next line it could be just `sequence` after parsing